src/main/java/cz/upce/fei/inptp/zz/injector/InstanceInjector.java

```diff
@@ -0,0 +1,40 @@
+package cz.upce.fei.inptp.zz.injector;
+
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import cz.upce.fei.inptp.zz.service.crypto.CryptoFileServiceModule;
+import cz.upce.fei.inptp.zz.service.json.JSONFileServiceModule;
+import cz.upce.fei.inptp.zz.service.password.PasswordDatabaseServiceModule;
+
+/**
+ * It provides the dependency injector for project.
+ *
+ * @author Ondřej Chrbolka
+ *
+ * */
+public final class InstanceInjector {
+    /**
+     * To save dependency single instance
+     * */
+    private static Injector injector;
+
+    private InstanceInjector(){
+    }
+
+    /**
+     * It will return the single instance of dependency injector.
+     *
+     * @return dependency injector.
+     * */
+    public static Injector injector(){
+        if(injector == null){
```

Comment:
possible threading problem, do not use lazy initialization

Reply:
Sorry about that. We used this method in OOP. I was searching on the internet how to path it and I found this link (https://stackoverflow.com/questions/8297705/how-to-implement-thread-safe-lazy-initialization) where the author had the same problem. So I used the correct solution to fix.